### DISCUSSION TITLE
예외 메세지를 테스트하는지, 한다면 어떻게 하는지 듣고 싶어요.

### DISCUSSION THREAD

**Discussion by nayonsoso** [2024-woowacourse-study/level-interview#72] [Test]
.

**Comment by alstn113** [comment]
예외 메세제를 테스트하는 것은 중요한 부분 중 하나입니다. 예외 메세지가 정확하게 발생하는지 확인하는 것은 코드의 신뢰성을 높이고 예기치 않은 동작을 방지하는데 도움이 됩니다.

예외 메세지를 테스트하는 방법의 예시는 다음과 같습니다.
```java
 assertThatThrownBy(
            () -> board.move(sourcePosition, targetPosition, Color.WHITE))
            .isInstanceOf(IllegalArgumentException.class)
            .hasMessage("선택한 기물이 존재하지 않습니다.");
```

예외 메세지를 테스트하기 위해서는 예외 메세지가 필요합니다.
예외 메세지는 특정 도메인에서 public 상수로 관리하고 밖에서 사용하거나 private 상수로 관리하고 테스트할 경우 하드 코딩할 수 있습니다.

보통의 경우 public 상수로 두고, 테스트 등에서 사용합니다.
예외 메세지에 값이나 상황 등등이 들어갈 경우 깨지기 쉬운 테스트를 만들기 위해서 하드 코딩할 것 같습니다.

**Comment by GIVEN53** [comment]
하나의 도메인에서 같은 타입의 예외 메시지를 던지는 부분이 많으면 예외 메시지까지 테스트한다.
커스텀 예외를 정의했다면 커스텀 예외 타입까지만 테스트해도 충분하다고 생각한다.
이 경우 `isExactlyInstanceOf(예외타입.class)`를 사용하여 상위 타입을 제외하고 명시한 타입과 일치하는지 확인한다.

예외 메시지가 중요한 도메인이면 하드 코딩해서 깨지기 쉬운 테스트를 만든다.
만약 예외 메시지를 enum으로 관리하고 있다면 enum 값으로 테스트한다.
